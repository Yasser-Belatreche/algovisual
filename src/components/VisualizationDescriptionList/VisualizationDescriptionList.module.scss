.container {
  @apply flex flex-col items-center gap-8;

  .descriptionItem {
    @apply flex items-start justify-around;

    &.reverse {
      @apply flex-row-reverse;
    }

    .picContainer {
      @apply w-4/12;
    }

    .text {
      @apply w-6/12 pt-28 flex flex-col items-start text-left justify-between gap-10;
      font-size: 20px;

      h2 {
        @apply text-blackShade leading-none font-bold;
        letter-spacing: -0.5px;
        font-size: 48px;
      }

      .desc {
        @apply text-grey w-10/12;
      }

      a {
        @apply text-blue font-bold;
      }
    }
  }

  @media only screen and (max-width: 1000px) {
    .descriptionItem {
      .text {
        h2 {
          font-size: 40px;
        }
      }
    }
  }

  @media only screen and (max-width: 800px) {
    @apply gap-16;

    .descriptionItem {
      @apply flex-col items-center gap-4;

      &.reverse {
        @apply flex-col;
      }

      .picContainer {
        @apply w-10/12;
        max-width: 350px;
      }

      .text {
        @apply w-full items-center text-center pt-0 gap-4;
        font-size: 18px;

        h2 {
          font-size: 35px;
        }

        .desc {
          @apply w-11/12;
        }
      }
    }
  }

  @media only screen and (max-width: 600px) {
    .descriptionItem {
      .text {
        h2 {
          font-size: 30px;
        }
      }
    }
  }

  @media only screen and (max-width: 400px) {
    .descriptionItem {
      .text {
        h2 {
          font-size: 26px;
        }
      }
    }
  }
}
