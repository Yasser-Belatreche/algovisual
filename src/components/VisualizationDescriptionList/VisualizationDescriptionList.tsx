import React, { useRef } from "react";

// styles
import styles from "./VisualizationDescriptionList.module.scss";

// components
import { DescriptionItem } from "./_components_/DescriptionItem";

export type Item = {
  title: string;
  description: string;
  pageLink: string;
  image: StaticImageData;
};

interface Props {
  listItems: Item[];
  firstItemTextPosition: "left" | "right";
}

const VisualizationDescriptionList: React.FC<Props> = ({
  listItems,
  firstItemTextPosition,
}) => {
  const offset = useRef<number>(1);

  if (firstItemTextPosition === "left") offset.current = 0;
  else offset.current = 1;

  return (
    <div className={styles.container}>
      {listItems.map((item, index) => {
        const isReversed = (index + offset.current) % 2 === 0;

        return <DescriptionItem key={index} {...item} reverse={isReversed} />;
      })}
    </div>
  );
};

export { VisualizationDescriptionList };
