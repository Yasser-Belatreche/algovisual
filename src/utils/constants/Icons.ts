import github from "../../../public/icons/github.svg";
import arrowDown from "../../../public/icons/arrow-down.svg";
import arrowLeft from "../../../public/icons/arrow-left.svg";
import code from "../../../public/icons/code.svg";

const ICONS = {
  github,
  arrowDown,
  arrowLeft,
  code,
};

export { ICONS };
