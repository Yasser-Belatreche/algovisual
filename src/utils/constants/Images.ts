import logo from "../../../public/images/Logo.png";
import heroPerson from "../../../public/images/hero-person.png";
import laptop from "../../../public/images/laptop.png";
import analytics from "../../../public/images/analytics.png";
import heroImage from "../../../public/images/hero-image.png";
import chart from "../../../public/images/Chart.png";
import screen from "../../../public/images/screen.png";
import tasks from "../../../public/images/tasks.png";
import sorting from "../../../public/images/sorting-image.png";

const IMAGES = {
  logo,
  heroPerson,
  laptop,
  analytics,
  heroImage,
  chart,
  screen,
  tasks,
  sorting,
};

export { IMAGES };
