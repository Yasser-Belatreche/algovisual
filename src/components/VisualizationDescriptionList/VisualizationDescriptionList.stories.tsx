import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { VisualizationDescriptionList } from "./VisualizationDescriptionList";
import { IMAGES } from "../../utils/constants/Images";

export default {
  title: "Global/VisualizationDescriptionList",
  component: VisualizationDescriptionList,
  argTypes: {},
} as ComponentMeta<typeof VisualizationDescriptionList>;

const Template: ComponentStory<typeof VisualizationDescriptionList> = (
  args
) => <VisualizationDescriptionList {...args} />;

const items = [
  {
    title: "Sorting Algorithms",
    description:
      "Explore how the most famous sorting algorithms works, we've included Bubble Sort, Merge Sort, Insertion Sort and more...",
    image: IMAGES.sorting,
    pageLink: "/visulizations/sorting",
  },
  {
    title: "Sorting Algorithms",
    description:
      "Explore how the most famous sorting algorithms works, we've included Bubble Sort, Merge Sort, Insertion Sort and more...",
    image: IMAGES.sorting,
    pageLink: "/visulizations/sorting",
  },
  {
    title: "Sorting Algorithms",
    description:
      "Explore how the most famous sorting algorithms works, we've included Bubble Sort, Merge Sort, Insertion Sort and more...",
    image: IMAGES.sorting,
    pageLink: "/visulizations/sorting",
  },
];

export const StartWithRightText = Template.bind({});
StartWithRightText.args = {
  listItems: items,
  firstItemTextPosition: "right",
};

export const StartWithLeftText = Template.bind({});
StartWithLeftText.args = {
  ...StartWithRightText.args,
  firstItemTextPosition: "left",
};
