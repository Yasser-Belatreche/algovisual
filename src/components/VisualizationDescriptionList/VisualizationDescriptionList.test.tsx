import { render } from "@testing-library/react";
import { IMAGES } from "../../utils/constants/Images";

import { VisualizationDescriptionList } from "./VisualizationDescriptionList";

describe("VisulizationDescriptionList compoenent", () => {
  const items = [
    {
      title: "Sorting Algorithms",
      description:
        "Explore how the most famous sorting algorithms works, we've included Bubble Sort, Merge Sort, Insertion Sort and more...",
      image: IMAGES.sorting,
      pageLink: "/visulizations/sorting",
    },
    {
      title: "Sorting Algorithms",
      description:
        "Explore how the most famous sorting algorithms works, we've included Bubble Sort, Merge Sort, Insertion Sort and more...",
      image: IMAGES.sorting,
      pageLink: "/visulizations/sorting",
    },
    {
      title: "Sorting Algorithms",
      description:
        "Explore how the most famous sorting algorithms works, we've included Bubble Sort, Merge Sort, Insertion Sort and more...",
      image: IMAGES.sorting,
      pageLink: "/visulizations/sorting",
    },
  ];

  it("should render a list of VisulizatoinDescriptions, by default the text in the first element is on the right side", () => {
    const { container } = render(
      <VisualizationDescriptionList
        listItems={items}
        firstItemTextPosition="right"
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("The text in the first element should be on the left side, when passing 'left' as value of the prop firstItemTextPosition", () => {
    const { container } = render(
      <VisualizationDescriptionList
        listItems={items}
        firstItemTextPosition="left"
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("the container should have the class 'containerClass' from the props", () => {
    const instance = render(
      <VisualizationDescriptionList
        listItems={items}
        firstItemTextPosition="left"
        containerClass="someClass"
      />
    );

    const container = instance.getByRole("listContainer");

    expect(container.classList.length).toEqual(2);
    expect(container.classList).toContain("someClass");

    expect(instance.container).toMatchSnapshot();
  });
});
