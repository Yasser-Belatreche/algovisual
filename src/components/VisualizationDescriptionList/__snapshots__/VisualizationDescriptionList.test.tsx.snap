// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisulizationDescriptionList compoenent The text in the first element should be on the left side, when passing 'left' as value of the prop firstItemTextPosition 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="descriptionItem reverse"
    >
      <div
        class="picContainer"
      >
        <img
          alt=""
          placeholder="blur"
          src="test-file-stub"
        />
      </div>
      <div
        class="text"
      >
        <h2>
           
          Sorting Algorithms
        </h2>
        <p
          class="desc"
        >
          Explore how the most famous sorting algorithms works, we've included Bubble Sort, Merge Sort, Insertion Sort and more...
        </p>
        <a
          href="/visulizations/sorting"
        >
          See Visualization
        </a>
      </div>
    </div>
    <div
      class="descriptionItem"
    >
      <div
        class="picContainer"
      >
        <img
          alt=""
          placeholder="blur"
          src="test-file-stub"
        />
      </div>
      <div
        class="text"
      >
        <h2>
           
          Sorting Algorithms
        </h2>
        <p
          class="desc"
        >
          Explore how the most famous sorting algorithms works, we've included Bubble Sort, Merge Sort, Insertion Sort and more...
        </p>
        <a
          href="/visulizations/sorting"
        >
          See Visualization
        </a>
      </div>
    </div>
    <div
      class="descriptionItem reverse"
    >
      <div
        class="picContainer"
      >
        <img
          alt=""
          placeholder="blur"
          src="test-file-stub"
        />
      </div>
      <div
        class="text"
      >
        <h2>
           
          Sorting Algorithms
        </h2>
        <p
          class="desc"
        >
          Explore how the most famous sorting algorithms works, we've included Bubble Sort, Merge Sort, Insertion Sort and more...
        </p>
        <a
          href="/visulizations/sorting"
        >
          See Visualization
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`VisulizationDescriptionList compoenent should render a list of VisulizatoinDescriptions, by default the text in the first element is on the right side 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="descriptionItem"
    >
      <div
        class="picContainer"
      >
        <img
          alt=""
          placeholder="blur"
          src="test-file-stub"
        />
      </div>
      <div
        class="text"
      >
        <h2>
           
          Sorting Algorithms
        </h2>
        <p
          class="desc"
        >
          Explore how the most famous sorting algorithms works, we've included Bubble Sort, Merge Sort, Insertion Sort and more...
        </p>
        <a
          href="/visulizations/sorting"
        >
          See Visualization
        </a>
      </div>
    </div>
    <div
      class="descriptionItem reverse"
    >
      <div
        class="picContainer"
      >
        <img
          alt=""
          placeholder="blur"
          src="test-file-stub"
        />
      </div>
      <div
        class="text"
      >
        <h2>
           
          Sorting Algorithms
        </h2>
        <p
          class="desc"
        >
          Explore how the most famous sorting algorithms works, we've included Bubble Sort, Merge Sort, Insertion Sort and more...
        </p>
        <a
          href="/visulizations/sorting"
        >
          See Visualization
        </a>
      </div>
    </div>
    <div
      class="descriptionItem"
    >
      <div
        class="picContainer"
      >
        <img
          alt=""
          placeholder="blur"
          src="test-file-stub"
        />
      </div>
      <div
        class="text"
      >
        <h2>
           
          Sorting Algorithms
        </h2>
        <p
          class="desc"
        >
          Explore how the most famous sorting algorithms works, we've included Bubble Sort, Merge Sort, Insertion Sort and more...
        </p>
        <a
          href="/visulizations/sorting"
        >
          See Visualization
        </a>
      </div>
    </div>
  </div>
</div>
`;
